libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1848,14 +1848,18 @@ void AP_InertialSensor::_save_gyro_calibration()
         _gyro_offset(i).save();
         _gyro_id(i).save();
 #if HAL_INS_TEMPERATURE_CAL_ENABLE
-        caltemp_gyro(i).save();
+        if (_backends[i]->externally_temperature_calibrated  == true) {
```

Comment:
This looks inverted?  Surely we only save if the thing isn't externall calibrated?